hrms/overrides/employee_advance.py

```diff
@@ -0,0 +1,239 @@
+                    frappe.throw(_(f"Invalid State Transition to state {self.state}"))
+                if not (
+                    frappe.session.user == self.approver_1
+                    or frappe.session.user == "Administrator"
+                ):
+                    frappe.throw(_("Only the Added approver can approve"))
+                CustomTask.mark_tasks_as_completed(
+                    "Employee Advance", self.name, TaskType.APPROVE_EMPLOYEE_ADVANCE
+                )
+            elif self.status == EmployeeAdvanceConstants.PENDING_PAYMENT:
+                if old_doc.status not in [EmployeeAdvanceConstants.PENDING_APPROVAL_BY_HR]:
+                    frappe.throw(_(f"Invalid State Transition to state {self.state}"))
+                if not user_has_role(frappe.session.user, RoleEnum.HR_L2_EXPENSE_ROLE.value):
+                    frappe.throw(
+                        _("Only user having HR L1 or L2 Expense Role can approve the document")
+                    )
+
+    def validate_approver(self):
+        user_id = frappe.get_list(
+            "Employee",
+            filters={"employee": self.employee},
+            fields=["user_id"],
+            pluck="user_id",
+        )[0]
+        if user_id == self.approver_1:
+            frappe.throw(_("Employee and its Approver cannot be the same"))
+
+        if self.advance_amount >= RoleEnum.ADVANCE_AMOUNT.value and not user_has_role(
+            self.approver_1, RoleEnum.EXPENSE_APPROVER2_ROLE.value
+        ):
+            frappe.throw(
+                _(
+                    f"Select L2 Level Approval since the amount is greater than or equal to {RoleEnum.ADVANCE_AMOUNT.value}"
+                )
+            )
+        elif not (
+            user_has_role(self.approver_1, RoleEnum.EXPENSE_APPROVER1_ROLE.value)
+            or user_has_role(self.approver_1, RoleEnum.EXPENSE_APPROVER2_ROLE.value)
+        ):
+            frappe.throw(
+                _(
+                    f"Approval selected does not have L1 or L2 Expense Approver Role since the amount is less than or equal to {RoleEnum.ADVANCE_AMOUNT.value}"
+                )
+            )
+
+    def update_claimed_amount(self):
+        claimed_amount = (
+            frappe.db.sql(
+                """
+            SELECT sum(ifnull(allocated_amount, 0))
+            FROM `tabExpense Claim Advance` eca, `tabExpense Claim` ec
+            WHERE
+                eca.employee_advance = %s
+                AND ec.name = eca.parent
+                AND ec.docstatus=1
+                AND eca.allocated_amount > 0
+        """,
+                self.name,
+            )[0][0]
+            or 0
+        )
+
+        employee_advance_doc = frappe.get_doc("Employee Advance", self.name)
+        employee_advance_doc.claimed_amount = claimed_amount
+        if employee_advance_doc.claimed_ammount == employee_advance_doc.sanctioned_amount:
+            employee_advance_doc.status = "Claimed"
```

Comment:
make these as status constants similar to rest of them

Reply:
Done